internal/providers/pluginfw/resources/user/data_users.go

```diff
@@ -0,0 +1,97 @@
+package user
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go/service/iam"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/converters"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/databricks/terraform-provider-databricks/internal/service/iam_tf"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+const dataSourceName = "users"
+
+func DataSourceUsers() datasource.DataSource {
+	return &UsersDataSource{}
+}
+
+var _ datasource.DataSourceWithConfigure = &UsersDataSource{}
+
+type UsersDataSource struct {
+	Client *common.DatabricksClient
+}
+
+type UsersInfo struct {
+	Filter types.String  `json:"filter,omitempty"`
```

Comment:
What I would recommend as well - by default fetch only limited number of attributes, i.e., don't include `roles` - this could be done to decrease the load on the SCIM backend as fetching roles is quite heavyweight operation. 

We can add an additional parameter, like, `extra_attributes` that will receive the list of additional attributes (comma-separated).

Reply:
Done! 👍